### DISCUSSION TITLE
WSL2 Support

### DISCUSSION THREAD

**Discussion by mwlock** [cyberbotics/webots#5879] [General]
Hey!

I am just wondering what the current of status is for running webots on WSL2?

The R2023a release notes mention
> Added support for Windows Subsystem for Linux (WSL) and macOS.

Although I am still having some trouble running on WSL where the simulation simply does not run once started (see the video)

https://user-images.githubusercontent.com/53016036/218152213-6f9890fb-8d4a-4557-87e4-590ce8e6b1e3.mp4

There are no error messages report in the terminal

**Comment by omichel** [comment]
It is simply not yet supported. Why do you need to run Webots from WSL2? Why can you run it from Windows?

**Reply by mwlock** [reply]
There's not an absolute need, but WSL is becoming increasingly convenient for me to quickly switch between different Linux environments 😄 I have Webots 2022a working just fine on WSL Ubuntu 22.04 and thought it would be nice to try Webots 2023a.

Can I ask what the releases are referring to then?

**Reply by omichel** [reply]
I can run Webots in WSL2 built from the [master](https://github.com/cyberbotics/webots) branch  of this repository. However, there are still some glitches, like I cannot close the guided tour window for some reason. But overall it works. If you want to investigate and fix these small glitches, you are welcome.

**Reply by mwlock** [reply]
I have built the master branch and it seems to be working! 😄
